spring-integration-core/src/main/java/org/springframework/integration/channel/MessagePublishingErrorHandler.java

```diff
@@ -140,8 +162,12 @@ public final void handleError(Throwable t) {
 	}
 
 	private MessageChannel resolveErrorChannel(Throwable t) {
-		Message<?> failedMessage = (t instanceof MessagingException) ?
-				((MessagingException) t).getFailedMessage() : null;
+		Throwable actualThrowable = t;
+		if (t instanceof OriginalMessageContainingMessagingException) {
+			actualThrowable = t.getCause();
+		}
+		Message<?> failedMessage = (actualThrowable instanceof MessagingException) ?
```

Comment:
Yes, but `t` might be any throwable (not in our code but we can't control user code).